sarray_mode.py

```diff
@@ -16,12 +19,51 @@ def mode_sa(sa, single_mode=True):
         max_count = sf2[max_count_index]["count"]
         return sf2[sf2["count"] == max_count]["value"]
 
+
 # Create an SArray with two modes (most-common elements: 2 and 3)
-# sa = gl.SArray([1, 2, 2, 3, 3])
+sa = gl.SArray([1, 2, 2, 3, 3])
 
 # Find one of the modes
-# print mode_sa(sa) # Returns 2
+single_mode = mode_sa(sa) # returns 2
 
 # Find all modes
-# print mode_sa(sa, single_mode=False) # Returns SArray with [2, 3]
+all_modes = mode_sa(sa, single_mode=False)
+# Returns
+# dtype: int
+# Rows: 2
+# [2, 3]
+
+
+# A faster (albeit maybe less accurate) way to find the mode value is using sa.sketch_summary().frequent_items() .
+# There are two caveats to this approach:
+# 1. won't work for very low-frequency mode values, and 
+# 2. won't necessarily give the correct result if there are multiple likely candidates.
+
+def sketch_mode_sa(sa, single_mode=True):
+    """Fast (albeit less accurate) way to find the mode value(s) of SArray sa.
+
+    single_mode: whether to return a single mode or an SArray of all modes (default: True)."""
+    
+    if len(sa) == 0:
+        raise ValueError("Can't find mode(s) in empty SArray")
+    
+    frequent_items_sketch = sa.sketch_summary().frequent_items()
+    modes_sketch = [k for (k, v) in frequent_items_sketch.iteritems()
```

Comment:
This may be slow if the list of frequent items is long (and there is not really an upper bound to how long it can be). Here it calls `itervalues` for each item of the dict and max on that. For perf this should be refactored to use a single call to max(itervalues) and a comparison to that stored value.